src/storage/engine/mod.rs

```diff
@@ -95,16 +96,11 @@ pub trait Snapshot: Send {
     fn get(&self, key: &Key) -> Result<Option<Value>>;
     fn get_cf(&self, cf: CfName, key: &Key) -> Result<Option<Value>>;
     #[allow(needless_lifetimes)]
-    fn iter<'a>(&'a self,
-                upper_bound: Option<&[u8]>,
-                fill_cache: bool,
-                mode: ScanMode)
-                -> Result<Cursor<'a>>;
+    fn iter<'a>(&'a self, iter_opt: IterOption, mode: ScanMode) -> Result<Cursor<'a>>;
```

Comment:
IMO `IterOption` contains attributes how to generate the iterator, and `ScanMode` is how to use this iterator.